lib/mapping/MapFormatH3M.h

```diff
@@ -245,7 +245,7 @@ class DLL_LINKAGE CMapLoaderH3M : public IMapLoader
 
 	/** List of templates loaded from the map, used on later stage to create
 	 *  objects but not needed for fully functional CMap */
-	std::vector<ObjectTemplate> templates;
+	std::vector<const ObjectTemplate*> templates;

```

Comment:
pointers on allocated objects are stored here. I suggest to use shared_ptr

Reply:
or unique_ptr, depends on the use case